src/replication/transfer/image/transfer.go

```diff
@@ -269,6 +272,14 @@ func (t *transfer) copyBlob(srcRepo, dstRepo, digest string) error {
 		return err
 	}
 	defer data.Close()
+
+	t.logger.Debugf("the blob size from api is %d, from manifests is %d", size, sizeFromDescriptor)
+	// get size 0 from PullBlob, use size from distribution.Descriptor instead.
+	if size == 0 {
+		size = sizeFromDescriptor
+		t.logger.Debugf("the blob size from api is 0, use size %d from manifests instead", size)
```

Comment:
`the blob size from api is 0, use size %d from manifests instead` to `the blob size from remote registry is 0, use size %d from manifests instead`